userspace/libsinsp/sinsp.cpp

```diff
@@ -1872,69 +1872,62 @@ void sinsp::init_mesos_client(string* api_server, bool verbose)
 	}
 }
 
-void sinsp::init_k8s_ssl(string* api_server, string* ssl_cert)
+void sinsp::init_k8s_ssl(const string &ssl_cert)
 {
 #ifdef HAS_CAPTURE
-	if(ssl_cert && (!m_k8s_ssl || ! m_k8s_bt))
+	if(!ssl_cert.empty() && (!m_k8s_ssl || ! m_k8s_bt))
 	{
 		std::string cert;
 		std::string key;
 		std::string key_pwd;
 		std::string ca_cert;
 
 		// -K <bt_file> | <cert_file>:<key_file[#password]>[:<ca_cert_file>]
-		std::string::size_type pos = ssl_cert->find(':');
+		std::string::size_type pos = ssl_cert.find(':');
 		if(pos == std::string::npos) // ca_cert-only is obsoleted, single entry is now bearer token
 		{
-			m_k8s_bt = std::make_shared<sinsp_bearer_token>(*ssl_cert);
-			ssl_cert->clear();
+			m_k8s_bt = std::make_shared<sinsp_bearer_token>(ssl_cert);
 		}
 		else
 		{
-			while(ssl_cert->length())
+			cert = ssl_cert.substr(0, pos);
+			if(cert.empty())
 			{
-				if(cert.empty() && pos != std::string::npos)
-				{
-					cert = ssl_cert->substr(0, pos);
-					if(ssl_cert->length() > (pos + 1))
-					{
-						*ssl_cert = ssl_cert->substr(pos + 1);
-					}
-					else { break; }
-				}
-				else if(key.empty())
-				{
-					key = ssl_cert->substr(0, pos);
-					if(ssl_cert->length() > (pos + 1))
-					{
-						*ssl_cert = ssl_cert->substr(pos + 1);
-						std::string::size_type s_pos = key.find('#');
-						if(s_pos != std::string::npos && key.length() > (s_pos + 1))
-						{
-							key_pwd = key.substr(s_pos + 1);
-							key = key.substr(0, s_pos);
-						}
-						if(pos == std::string::npos) { break; }
-					}
-					else { break; }
-				}
-				else if(ca_cert.empty())
-				{
-					ca_cert = *ssl_cert;
-					ssl_cert->clear();
-				}
-				else { goto ssl_err; }
-				pos = ssl_cert->find(':', pos);
+				throw sinsp_exception(string("Invalid K8S SSL entry: ") + ssl_cert);
+			}
+
+			// pos < ssl_cert.length() so it's safe to take
+			// substr() from head, but it may be empty
+			std::string::size_type head = pos + 1;
+			pos = ssl_cert.find(':', head);
+			if (pos == std::string::npos)
+			{
+				key = ssl_cert.substr(head, std::string::npos);
```

Comment:
nit: if string::npos is being passed for clarity, why not on line 1910 as well? i'd just get rid of it.

Reply:
Yup, I'll remove it. It was a testing leftover that should have been cleaned up.